owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/ShareFileComposeFragment.kt

```diff
@@ -0,0 +1,626 @@
+
+                // Get capabilities to update some UI elements depending on them
+                val capabilitiesState = ocCapabilityViewModel.capabilities.observeAsState()
+                val capabilitiesUiResult = capabilitiesState.value?.peekContent()
+                val capabilities = capabilitiesUiResult?.getStoredData()
+                val isShareApiEnabled = capabilities?.filesSharingApiEnabled == CapabilityBooleanType.TRUE
+                val isPublicShareEnabled = capabilities?.filesSharingPublicEnabled == CapabilityBooleanType.TRUE
+                val isMultiplePublicSharingEnabled = capabilities?.filesSharingPublicMultiple?.isTrue ?: false
+
+                // Get shares to update some UI elements depending on them
+                val sharesState = ocShareViewModel.shares.observeAsState()
+                val sharesUiResult = sharesState.value?.peekContent()
+                val shares = sharesUiResult?.getStoredData()
+                var privateShares = shares?.filter { share ->
+                    share.shareType == ShareType.USER ||
+                            share.shareType == ShareType.GROUP ||
+                            share.shareType == ShareType.FEDERATED
+                }
+                var publicShares = shares?.filter { share ->
+                    share.shareType == ShareType.PUBLIC_LINK
+                }
+
+                when (capabilitiesUiResult) {
+                    is UIResult.Success -> {
+                        listener?.dismissLoading()
+                    }
+                    is UIResult.Error -> {
+                        capabilitiesState.value?.getContentIfNotHandled()?.let {
+                            showErrorInSnackbar(R.string.get_capabilities_error, capabilitiesUiResult.error)
+                        }
+                        listener?.dismissLoading()
+                    }
+                    is UIResult.Loading -> {
+                        listener?.showLoading()
+                    }
+                    else -> {
+                        // To avoid non-exhaustive when warning
+                    }
+                }
+
+                when (sharesUiResult) {
+                    is UIResult.Success -> {
+                        listener?.dismissLoading()
+                    }
+                    is UIResult.Error -> {
+                        sharesState.value?.getContentIfNotHandled()?.let {
+                            showErrorInSnackbar(R.string.get_shares_error, sharesUiResult.error)
+                        }
+                        listener?.dismissLoading()
+                    }
+                    is UIResult.Loading -> {
+                        listener?.showLoading()
+                    }
+                    else -> {
+                        // To avoid non-exhaustive when warning
+                    }
+                }
+
+                LazyColumn {
+                    item { SharedFileRow() }
```

Comment:
I understand why you did it this way, and it's totally fine. But, probably it would be a good idea to send the file here instead of accessing a global variable within the composable. That way, we could maintain previews, or we could test the composable by itself.

Something like:
```suggestion
                    item { SharedFileRow(ocFile) }
```

So, totally up to you 😉

Reply:
Nice, I think it's a good change. I'll apply it!